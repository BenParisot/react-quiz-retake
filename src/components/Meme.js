import React, { PureComponent } from 'react';
import TextSelector from './TextSelector';
import Display from './Display';

export default class Meme extends PureComponent {
    state = {
      headerText: '',
      footerText: '',
      imageSrc: ''
    }

    handleHeaderChange = ({ target }) => {
      this.setState({ headerText: target.value });
    }

    handleFooterChange = ({ target }) => {
      this.setState({ footerText: target.value });
    }

    handleImageSrcChange = ({ target }) => {
      this.setState({ imageSrc: target.value });
    }

    render() {
      const { headerText, footerText, imageSrc } = this.state;
      return (
          <>
        <TextSelector handleHeaderChange={this.handleHeaderChange} handleFooterChange={this.handleFooterChange} handleImageSrcChange={this.handleImageSrcChange}/>
        <Display headerText={headerText} footerText={footerText} imageSrc={imageSrc} />
        </>
      );
    }

}
