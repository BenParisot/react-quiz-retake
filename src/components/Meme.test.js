import React from 'react';
import { shallow } from 'enzyme';
import Meme from './Meme';

describe('Meme component', () => {
  it('displays a meme', () => {
    const wrapper = shallow(<Meme />);
    expect(wrapper).toMatchSnapshot(); 
  });
});
