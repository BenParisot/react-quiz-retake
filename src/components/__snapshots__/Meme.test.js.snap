// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meme component displays a meme 1`] = `
<Fragment>
  <TextSelector
    handleFooterChange={[Function]}
    handleHeaderChange={[Function]}
    handleImageSrcChange={[Function]}
  />
  <Display
    footerText=""
    headerText=""
    imageSrc=""
  />
</Fragment>
`;
